components/builder-worker/src/studio.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::process::Command;
+use std::path::Path;
+
+use protocol::jobsrv as proto;
+
+use error::Result;
+
+pub fn build(job: &proto::Job, root: &Path) -> Result<()> {
+    let output = Command::new("sh")
```

Comment:
I know there's work to do here, but here are some to-implements that occur to me:

* fork a child process with some upper bounds timeout
* ensure that the the forked process user is root or user with sufficient perms to bind mount, dev, etc
* skip the unwrap so we get a clean pass/fail and not a dead thread
* put the forked process into the correct `$workspace/$vcs_dir` directory before issuing the build command (this determines the `/src` bind mount dir root inside the studio
* run the process either in a shell session with a tty assigned or ensure that the parent is doing the same. we'll make `hab` better for this but can't control other software that may need this ([related](https://github.com/travis-ci/travis-ci/issues/1337))